src/main/java/gov/nasa/jpl/mbee/mdk/systems_reasoner/actions/SpecializeStructureRecursiveAndIndividualAction.java

```diff
@@ -0,0 +1,145 @@
+package gov.nasa.jpl.mbee.mdk.systems_reasoner.actions;
+
+import com.nomagic.magicdraw.copypaste.CopyPasting;
+import com.nomagic.magicdraw.core.Application;
+import com.nomagic.magicdraw.openapi.uml.SessionManager;
+import com.nomagic.magicdraw.ui.dialogs.MDDialogParentProvider;
+import com.nomagic.magicdraw.ui.dialogs.SelectElementInfo;
+import com.nomagic.magicdraw.ui.dialogs.SelectElementTypes;
+import com.nomagic.magicdraw.ui.dialogs.selection.ElementSelectionDlg;
+import com.nomagic.magicdraw.ui.dialogs.selection.ElementSelectionDlgFactory;
+import com.nomagic.magicdraw.ui.dialogs.selection.SelectionMode;
+import com.nomagic.uml2.ext.jmi.helpers.ModelHelper;
+import com.nomagic.uml2.ext.magicdraw.auxiliaryconstructs.mdmodels.Model;
+import com.nomagic.uml2.ext.magicdraw.classes.mdkernel.*;
+import gov.nasa.jpl.mbee.mdk.docgen.validation.ValidationRuleViolation;
+import gov.nasa.jpl.mbee.mdk.lib.Utils;
+import gov.nasa.jpl.mbee.mdk.validation.actions.AddInheritanceToAssociationAction;
+import gov.nasa.jpl.mbee.mdk.validation.actions.RedefineAttributeAction;
+
+import java.awt.*;
+import java.awt.event.ActionEvent;
+import java.util.ArrayList;
+import java.util.List;
+
+public class SpecializeStructureRecursiveAndIndividualAction extends SRAction {
```

Comment:
Having three classes that do very similar things may not be the best option from a maintainability perspective. Is there any opportunity to consolidate these actions and differ behavior based on a constructor argument?